### ISSUE TITLE
Audit clothing crafting materials and dissasembly

### ISSUE THREAD

**Issue by Terrorforge** [CleverRaven/Cataclysm-DDA#53059]
### Is your feature request related to a problem? Please describe.

The clothing material situation is kind of a hot mess right now. There's a confusing number of different types of textile components, limited ways to acquire those components, and little consistency between or even within each material type in how they're used. Here are just some of the issues:

**Cotton**

- Cotton has no less than 4 different types of material associated with it; rags, cotton sheets, patchwork cotton clothing parts, and patchwork cotton sheets.
- Recipes will use rags or sheets practically at random. If they use sheets, they will sometimes allow patchwork cotton sheets instead, but not always. 
- If you do need cotton sheets, they're quite annoying to acquire, because other than getting lucky and finding a bunch in a mansion or clothing store, you can only get them from disassembling a very specific list of arbitrary items. E.g. you can take apart a pair of denim shorts for cotton sheets, but not a pair of jeans. 
- Patchwork cotton clothing parts aren't used for anything except repairing, making them kind of redundant, except oh no actually they're used in exactly three recipes - the two nomad jumpsuits and the pair of sneakers.

**Nomex**

- Nomex has three materials; Nomex patches, Nomex sheets and Nomex patchwork clothing parts. 
- The patchwork clothing parts seem to serve the same purpose as the patchwork cotton sheet, i.e. as a replacement for proper sheets, but unlike the patchwork cotton clothing part, aren't used for repairs.
- The ONLY way to get Nomex sheets is to disassemble turnout boots. 
- You can use Nomex as a replacement for cotton in a lot of regular clothing items like the bathrobe, and in those cases you will used the same number of items regardless of whether they're patches, sheets or clothing parts, despite the fact that one clothing part takes five patches to craft, and we can tell from recipes like the survivor fireboots that one clothing part is equivalent to something like 20 sheets.

**Kevlar**
- Doesn't quite have the confusion problem since kevlar sheets, layered kevlar panels and rigid kevlar plates are distinctly different materials, but suffers from a serious lack of sources.
- Other than butchering kevlar zombies, kevlar sheets can pretty much only be acquired from combat boots and a few rare items like cut-resistant gloves.
- The only practical source of layered kevlar panels is salvaging kevlar vests.
- Rigid kevlar plates can only be salvaged from carbon fiber boat hulls and carbon fiver fishing spears (?). But then they don't seem to be used in any crafting, so.
- The root problem seems to be a lack of disassembly recipes. There are plenty of items that are made of kevlar or layered kevlar, but the vast majority of them simply cannot be salvaged or dissasembled.

**Lycra**
- Again, three different materials (patch, sheet, patchwork clothing part) with seemingly no rhyme or reason to which can be used where. Sometimes they're interchangeable (zentai), sometimes a sheet or patchwork part will do (Nomad bodyglove), sometimes it has to be sheets (belly band holster)
- The only way to get Lycra sheets is once again to get lucky and find it in a mall, mansion or clothing store, or disassemble one of exactly two items: leggings and bikini bottom.

**Summary**
Overall, the two main problems seem to be that a lot of a items lack disassembly/salvage recipes, and that the profusion of various different varieties of each material has caused a lot of confusion about which versions are supposed to be used where, resulting in a lot of inconsistency as different contributors have used them differently.

### Solution you would like.

Honestly I think we should go back to just having a single "component" item for each material type. If we're absolutely determined to mark the difference between hastily shredded patches and nice, contiguous sheets, make it exactly TWO types, patches and sheets, and just let players craft sheets directly instead of using this awkward "patchwork clothing parts" workaround.

Even if you disagree on that being the right solution, we do need to do something to the fundamental structure of this to make it clearer. We'll need to go in and sort of the recipes and disassemblies either way, but if we don't also reorganize this to be more comprehensible contributor will continue to get confused and introduce weird, incongruous recipes.

**Comment by mmiki2305** [comment]
I find that a good source of Kevlar sheets is cutting up turnout gear, but other than that, I completely agree.  I understand what was the idea behind moving to the current system, but it just ended up being very confusing and inconsistent.

**Comment by Terrorforge** [comment]
> I find that a good source of Kevlar sheets is cutting up turnout gear

Hm, so you can. It's not listed on HHG, which I was using to check. I thought it was accurate, because it doesn't list the kevlar dog harness and I know you can't cut that up, but on closer inspection that specific item just has a NO_SALVAGE tag for some reason.

So the source situation isn't quite as bad as I thought, at least not for kevlar. But it remains a problem for other materials, since they generally salvage into patches, when it's the sheets you need.

**Comment by Maleclypse** [comment]
I think it would be nice if we had light industry variants that represented specialty clothing factories. I know of at least one in Massachusetts that made clothing in country for the DoD. That would also help with every light industry having welding gear.

**Comment by Photoloss** [comment]
Tangentially related problem: the absurdity of "cut up" vs "disassemble". I can "cut up" a pair of fur hot pants into 1 fur pelt in 0 seconds. A fur loincloth needs to be disassembled over 15 minutes to yield 2 fur pelts, cutting it up is not an option. And the fur bikini top takes **one freaking hour!** to disassemble into 4 fur pelts and some thread, again with no option of just cutting it up. The same applies to seemingly random cotton and lycra clothing.

**Comment by Scorch-Mechanic** [comment]
You can get rigid kevlar plates from cutting up army helmets (because that's one of their materials).

I'm also of the opinion that the phrase "patchwork [material] clothing parts" is has too many syllables and is too-complex a name for what is functionally a rag. I acknowledge that the majority of the dev team wants to obsolete the latter and switch entirely to the former, even though I question the point of creating a whole new item instead of just renaming the old one. 

In addition, I advocate strongly for changing the name of the new item to "patchwork [material] cloth" or "patchwork [material] sheet", in order to convey the same information concisely.

I recognize that patchwork cloth is supposed to be 20 times more massive than sheets, but there's at least one recipe I encountered recently that requires both for some reason, despite having so little final mass that it couldn't possibly be made out of more than one of the patchwork cloth items, and that's not even accounting for the mass of the other material requirements. The recipe is the ankle ammo pouch, which somehow has a volume five times that of a patchwork synthetic cloth item while having a mass ten grams less than one of the same. I realize that the interior volume of an empty pouch is different than the volume of the cloth making it, but I just don't see how that math can work.

**Comment by PatrikLundell** [comment]
@Photoloss The reason for the bikini top taking a lot of time to disassemble is obvious: the item is an extremely delicate balance between covering as little as possible while ensuring it's (usually) covering enough. As such, cutting the wrong strand results in not getting anything at all, and there are many strands to cut, so the work is similar to that of defusing a bomb ;P

**Comment by Photoloss** [comment]
> @Photoloss The reason for the bikini top taking a lot of time to disassemble is obvious: the item is an extremely delicate balance between covering as little as possible while ensuring it's (usually) covering enough. As such, cutting the wrong strand results in not getting anything at all, and there are many strands to cut, so the work is similar to that of defusing a bomb ;P

Does the corresponding "educational" material not exist in the C:DDA world? You'd think all those SD cards and USB drives would be full of it...

**Comment by PatrikLundell** [comment]
I'm not aware of any "educational" version that focuses on recovery of materials from the clothing shed.
Also note that the Cataclysm is focused on an area in the US, so all the "educational" media was probably destroyed by mobs who had their priorities set during the rioting phase of the Cataclysm.

**Comment by SpeedyDoggo** [comment]
There's also a bit where you can get rags from patchwork cotton clothing parts but not vice versa. Given that rags are used in creating bandages and other recipes that aren't necessarily clothing, I'd be all for folding the clothing parts into rags and maybe renaming them 'cotton rags', and use those for clothing repair instead when repairing cotton clothing.
(and would personally love to hop onto a project that audits the crafting/disassembly recipes to help with this change or just make sure things are complete)

**Comment by Scorch-Mechanic** [comment]
I've looked at the comments from the contributor who started the process with the patchwork parts, and it's actually the reverse: rags have been in for ages, and they're supposed to be replaced eventually with patchwork parts. But that work remains perhaps half-complete, because there are many recipes and apparently some hardcoded things that either use or produce rags. The recipe to turn patchwork parts into rags is supposed to be a stopgap until rags are obsoleted.

**Comment by Zireael07** [comment]
@Scorch-Mechanic  @SpeedyDoggo  yep, rags were here first and the move to patchwork wasn't completed.

**Comment by Terrorforge** [comment]
I had a dig around, and it looks like at some point every material got a "sheet_[material]" and "sheet_[material]_patchwork", with the in-game names "[material] sheet" and "patchwork [material] clothing part", defined in tailoring.json. This includes cotton.

Then #50207 comes in to fix certain problems caused by this. They notice that all these materials are defined as AMMO*, which is causing problems such as cutting up filthy clothes not resulting in filthy parts. So their solution is to rename sheet_cotton_patchwork from "patchwork cotton clothing parts" to "patchwork cotton sheet"** and add a new item with the ID "cotton_patchwork"*** in toiletries.json called "patchwork cotton clothing parts". So that's where that inconsistency comes from.

Another cause of inconsistency is that the sheet_material items are ridiculously tiny. Idk about you guys, but when I hear "sheet", I think of a large contiguous piece of material. This is about in line with the sheet_material_patchwork items, e.g. sheet_cotton_patchwork weighs 100g which works out to about 1 square meter of cotton fabric - but sheet_cotton is only 5 grams, which is a square of fabric about 22x22cm. That's a patch! In fact, it's significantly smaller than the actual "patch" items. Why is the "sheet" item so tiny and divisible? More importantly, if the only purpose of the patchwork sheets is so you have a way of meeting sheet requirements when you only have rags, why are they so different from the actual sheets? Actually, why do they exist at all? As I said at the top, if patchwork sheets are just replacements for sheets, why not just let players craft sheets directly?

* Presumably so they have the stacking behavior of ammo, i.e. in advanced inventory you just hit enter to pick up the whole stack as opposed to having to hit M like you do with rafs
** They should all be named like this tbh. "patchwork [material] clothing parts" is an extremely clumsy phrasing
*** Very confusing nomenclature, which is presumably why

**Comment by terribleperson** [comment]
I don't want to see the removal of the multiple component items for each material type, though I do think they could use some clarification. Multiple component items per material is consistent with other parts of the game (wood with 4, steel with 3) and has gameplay and realism benefits. Most of the fabric component items are or can be well-justified at least to my mind. 

- Sheets represent the standard form of a fabric, as recipes (in the case of IRL sewing, patterns) expect it. That is, standard-width pieces of fabric with two parallel edges cut to length off of a roll or long enough to fit most patterns.  I'm not sure but I think the current unit for sheet might be linear cm? Sheets are important, because sewing patterns are designed to be cut out of and stitched together from standard width fabric.

- To contrast this, patchwork sheets are a large contiguous piece of material your survivor has constructed out of scrap to turn unusable materials into something suitable for patterns designed for standard-width fabric. 
- Patches are neccesary because most items are not large enough or shaped properly to recover entire sheets of fabric. Making them return entire sheets of fabric anyways would reduce realism and make the materials too common, but not returning anything makes enormous numbers of items essentially worthless. 
- Rags are a necessary evil for now as others have said.

That all said, there are several adjustments I'd personally like to see, of varying difficulty and worth.

1. The different fabric component items need to be made consistent in name, behavior, and purpose. This would make things easier on both players and contributors
	- Patches/rags should have unified naming that fits their use and is consistent with each other and the rest of the game. I suggest "scrap of [material]", e.g. "scrap of cotton fabric", "scrap of lycra fabric". If that doesn't fit the naming of items in game, "cotton scrap", "lycra scrap" would be fine. I'm not certain that the current fabric scrap items justify their existence.
	- Patch/scrap pieces should all represent the same area of fabric sheet and be convertible to patchwork sheets by similar or identical recipes. This area could be reduced to cut down on the number of clothing items that can't return any materials if that's still an issue.
	- Patchwork sheets should represent the same area of fabric sheet per unit as a standard sheet and take up the same or slightly more weight and volume as the standard equivalent. Patchwork sheets are constructed from scrap, so they might involve slightly more material for the same area.
	- Recipes that accept sheet should accept patchwork sheet by default. A recipe could opt out of allowing patchwork sheet via some sort of "can't be patchwork" flag.
2. Working with patchwork material could be made more relevant to gameplay. Crafting patchwork sheets already acts as a time penalty to using patchwork materials, but it could have more far-reaching effects.
	- Using patchwork sheet for crafting could reasonably have a time and/or skill penalty for crafting. It might not lay or fold completely perfect, so that's going to make crafting somewhat slower and more difficult though not prohibitively so. Considering that, a time penalty when patchwork sheet is being used as a component is probably appropriate. The item already says there is one, but I haven't checked on it.
	- A 'patchwork material' flag could be added. Items crafted from patchwork sheet would have this flag turned on. Repairing items could potentially add this flag with a chance that depends on skill and the material used, scaling with the % of durability being repaired (that's always the same amount last I played, but I could see that changing). This flag could potentially have various effects e.g.  slightly reduced durability or very slightly increased weight. If your outer clothing was patchwork this could be visible to NPCs, although what they'd make of that information I don't know. Patchwork 'fancy' items might not provide a bonus for the 'stylish' trait. Patchwork outer clothing on NPCs could be visible to you, maybe letting you know that this person who says they're a soldier who luckily avoided all the ruckus of the cataclysm isn't being completely honest about where they got that uniform or that this "brand new, never before used!" kevlar vest isn't quite the deal it appears.

**Comment by Photoloss** [comment]
> Rags are a necessary evil for now as others have said.

What I don't understand is why the new pieces need to exist, instead of "rags" being renamed to match the new conventions. Just the desire to unify the invisible item IDs, even though anyone working with them will need to look them up anyway? ("scrap_cotton" vs "cotton_scrap" vs "patch_cotton"...)

> Patchwork sheets should represent the same area of fabric sheet per unit as a standard sheet and take up the same or slightly more weight and volume as the standard equivalent. Patchwork sheets are constructed from scrap, so they might involve slightly more material for the same area.

*Without* the second major point there is no reason to do this. Just have the crafting recipe for the sheet consume more material than the resulting item contains.

Basically it feels like "patchwork" versions should only have been introduced by a future PR implementing the slightly weaker "patchwork" flag/material types and have no reason to exist currently. They don't add realism due to lack of consequences, they are very unfun to play with and they clearly haven't managed to fix any issues surrounding "rags".

**Comment by terribleperson** [comment]
> _Without_ the second major point there is no reason to do this. Just have the crafting recipe for the sheet consume more material than the resulting item contains.

Sorry, I wrote that post yesterday and missed some stuff when I copied my draft into the browser and did a final pass. Patchwork sheets currently serve a purpose in that they make it possible for some recipes to only accept 'proper' materials, rather than salvaged ones. 

Ideally, the rag/patchwork cotton clothing part issue will go away on its own as fewer items return rags and they're used in less and less places, just like the old batteries went away.

**Comment by Terrorforge** [comment]
> Sorry, I wrote that post yesterday and missed some stuff when I copied my draft into the browser and did a final pass. Patchwork sheets currently serve a purpose in that they make it possible for some recipes to only accept 'proper' materials, rather than salvaged ones.

At the moment that's a downside, because it just exacerbates the other problems of inconsistent recipes and limited access to proper sheets.

**Comment by a-chancey** [comment]
I mentioned it over [here](https://github.com/CleverRaven/Cataclysm-DDA/issues/53315#issuecomment-991080436) on #53315, but we're probably going to need to add patchwork kevlar and kevlar patches or scraps as part of this. There's some odd deconstruction with Combat and Motorcycle boots - on top of me having qualms with the items themselves.

**Comment by a-chancey** [comment]
This should be good to close when #54256 is merged if nobody has objections.

**Comment by Terrorforge** [comment]
Sounds right to me. Thank you for all your hard work, this one was a doozy.
